Tests/Shared/Core/UnitTestBase.swift

```diff
@@ -112,6 +112,9 @@ open class UnitTestBase<Component>: XCTestCase {
 
     // All data that the SDK writes, delete it here so each test function has a clean environment to run and does not depend on the result of the previous test.
     open func deleteAllPersistentData() {
+        // need to remove the observers for integration tests that utilizes actual NotificationCenter
+        // otherwise, results are flaky
+        Task { await diGraphShared.eventBusHandler.removeAllObservers() }
```

Comment:
This change will probably result in flaky test results. 

When you create a `Task{}` we do not know when the task will execute. There is a chance that `diGraphShared.eventBusHandler.removeAllObservers()` will happen after tearDown is done running and so it can't be reliable. We need a guarantee that `removeAllObservers` will be done executing before tearDown() is done running. 

Ideas to fix this: 
* Make `eventBusHandler.removeAllObservers()` not an `async` function. Make it sync.
* Modify EventBusHandler and EventBus `deinit` function to call `removeAllObservers`. Then, when our test suite runs DIGraph.reset(), the `removeAllObservers` should be called for us, no longer needing this line of code in our tearDown method at all. 
* There are other options such as making `deleteAllPersistentData` an `async` function but that would be a much larger change. I don't suggest doing that at this time.